Sources/Context/Context.swift

```diff
@@ -0,0 +1,54 @@
+public protocol Context {
+    static var background: Context { get }
+    static func TODO(_ reason: StaticString?, function: StaticString, file: StaticString, line: UInt) -> Context
+    
+    subscript<Key: ContextValueContainerKey>(_ key: Key.Type) -> Key.Value? { get set }
+    
+    func forEach(_ body: (AnyContextValueContainerKey, Any) throws -> Void) rethrows
+}
+
+// MARK: Default implementations
+
+extension Context {
+    public static var background: Context { return EmptyContext.background }
+    
+    public static func TODO(_ reason: StaticString?, function: StaticString, file: StaticString, line: UInt) -> Context {
+        return EmptyContext.TODO(reason, function: function, file: file, line: line)
+    }
+    
+    public subscript<Key: ContextValueContainerKey>(_ key: Key.Type) -> Key.Value? {
+        get { return nil }
+        set { }
+    }
+    
+    public func forEach(_ body: (AnyContextValueContainerKey, Any) throws -> Void) rethrows { }
```

Comment:
these absolutely should not have default implementations -- a context/baggage MUST implement those correctly.

Both the forEach and the subscript, they must not be left to accidentally miss implementing them.